system/jlib/jsocket.cpp

```diff
@@ -4588,12 +4588,12 @@ class CSocketEpollThread: public CSocketBaseThread
             SelectItem *si = items.element(i);
             if (si->del)
             {
-                si->nfy->Release();
                 try
                 {
 #ifdef SOCKTRACE
                     PROGLOG("CSocketEpollThread::updateItems release %d",si->handle);
 #endif
+                    si->nfy->Release();
```

Comment:
similar to comment below (and not really new either).. wrapping ->Release() calls implies these might throw exceptions in their destructors.. they really shouldn't.
If  nfy did it would leak sock here.